code/modules/religion/sparring/sparring_datum.dm

```diff
@@ -0,0 +1,293 @@
+
+/datum/sparring_match/proc/hulk_interference(datum/source, mob/attacker)
+	SIGNAL_HANDLER
+	if((attacker == chaplain || attacker == opponent))
+		// fist fighting a hulk is so dumb. i can't fathom why you would do this.
+		return
+	flub(attacker)
+
+/datum/sparring_match/proc/hand_interference(datum/source, mob/living/attacker)
+	SIGNAL_HANDLER
+	if(!attacker.combat_mode)
+		//not worth the flub
+		return
+	if(attacker == chaplain || attacker == opponent)
+		//you can pretty much always use fists as a participant
+		return
+
+	flub(attacker)
+
+/datum/sparring_match/proc/paw_interference(datum/source, mob/living/attacker)
+	SIGNAL_HANDLER
+
+	if(attacker == chaplain || attacker == opponent)
+		//you can pretty much always use paws as a participant
+		return
+
+	flub(attacker)
+
+/datum/sparring_match/proc/thrown_interference(datum/source, atom/movable/thrown_movable, skipcatch = FALSE, hitpush = TRUE, blocked = FALSE, datum/thrownthing/throwingdatum)
+	SIGNAL_HANDLER
+	if(istype(thrown_movable, /obj/item))
+		var/mob/living/honorbound = source
+		var/obj/item/thrown_item = thrown_movable
+		var/mob/thrown_by = thrown_item.thrownby?.resolve()
+		if(thrown_item.throwforce < honorbound.health && ishuman(thrown_by))
+			flub(thrown_by)
+
+/datum/sparring_match/proc/projectile_interference(datum/participant, obj/projectile/proj)
+	SIGNAL_HANDLER
+	if(proj.firer == chaplain || proj.firer == opponent)
+		//oh, well that's allowed. or maybe it isn't. doesn't matter because firing the gun will trigger a violation, so no additional violation needed
+		return
+	var/mob/living/interfering
+	if(isliving(proj.firer))
+		interfering = proj.firer
+	flub(interfering)
+
+///someone randomly fucking died
+/datum/sparring_match/proc/death_flub(datum/deceased)
+	SIGNAL_HANDLER
+
+	flubbed_match()
+
+///someone randomly fucking deleted
+/datum/sparring_match/proc/deletion_flub(datum/qdeleting)
+	SIGNAL_HANDLER
+
+	flubbed_match()
+
+///someone used a gun
+/datum/sparring_match/proc/gun_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using guns")
+
+///someone used a grenade
+/datum/sparring_match/proc/grenade_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using grenades")
+
+///someone used melee weapons
+/datum/sparring_match/proc/melee_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using melee weapons")
+
+/datum/sparring_match/proc/teleport_violation(datum/offender)
+	SIGNAL_HANDLER
+	if(offender == chaplain)
+		end_match(opponent, chaplain, violation_victory = TRUE)
+	else
+		end_match(chaplain, opponent, violation_victory = TRUE)
+
+///someone tried to leave
+/datum/sparring_match/proc/arena_violation(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+
+	var/area/inhabited_area = get_area(mover)
+	if(inhabited_area == arena_condition)
+		return //still in the ring!! :)
+
+	violation(mover, "leaving the arena")
+	var/atom/throw_target = get_edge_target_turf(mover, REVERSE_DIR(direction))
```

Comment:
are you sure that this will fling them back into the arena? is direction based on the direction of the moving creature? if so, if they get shoved out of the ring while facing the center of it, the throw here could fling them FURTHER out of the ring.

Reply:
This is the direction not of the person but the direction they left the area, it should be pretty good at doing what it does.